### ISSUE TITLE
validation wrong

### ISSUE THREAD

**Issue by leigaoxiang** [clovaai/deep-text-recognition-benchmark#127]
when the code run at      valid_loss, current_accuracy, current_norm_ED, preds, confidence_score, labels, infer_time, length_of_data = validation(model, criterion, valid_loader, converter, opt) , it stop here and raise no errors, Program cannot continue

**Comment by ku21fan** [comment]
Hello, 

Check your GPU usage.
If there is no error, the program would be still running.
It may take much time depending on the GPU.

You can check faster with the additional commands
`--valInterval 10` or `--batch_size 16` 

Best.

**Comment by leigaoxiang** [comment]
but when i run train.py ,my computer quto stop, and then i comment validate part, it can run,is something wrong with validate

**Comment by leigaoxiang** [comment]
# validation part
        if i % opt.valInterval == 0:
            elapsed_time = time.time() - start_time
            # for log

            # with open(f'./saved_models/{opt.experiment_name}/log_train.txt', 'a') as log:
            #     model.eval()
            #     with torch.no_grad():
            #         valid_loss, current_accuracy, current_norm_ED, preds, confidence_score, labels, infer_time, length_of_data = validation(
            #             model, criterion, valid_loader, converter, opt)
            #     model.train()
            #
            #     # training loss and validation loss
            #     loss_log = f'[{i}/{opt.num_iter}] Train loss: {loss_avg.val():0.5f}, Valid loss: {valid_loss:0.5f}, Elapsed_time: {elapsed_time:0.5f}'
            #     print(loss_log)
            #     log.write(loss_log + '\n')
            #     loss_avg.reset()
            #
            #     current_model_log = f'{"Current_accuracy":17s}: {current_accuracy:0.3f}, {"Current_norm_ED":17s}: {current_norm_ED:0.2f}'
            #     print(current_model_log)
            #     log.write(current_model_log + '\n')
            #
            #     # keep best accuracy model (on valid dataset)
            #     if current_accuracy > best_accuracy:
            #         best_accuracy = current_accuracy
            #         torch.save(model.state_dict(), f'./saved_models/{opt.experiment_name}/best_accuracy.pth')
            #     if current_norm_ED < best_norm_ED:
            #         best_norm_ED = current_norm_ED
            #         torch.save(model.state_dict(), f'./saved_models/{opt.experiment_name}/best_norm_ED.pth')
            #     best_model_log = f'{"Best_accuracy":17s}: {best_accuracy:0.3f}, {"Best_norm_ED":17s}: {best_norm_ED:0.2f}'
            #     print(best_model_log)
            #     log.write(best_model_log + '\n')
            #
            #     # show some predicted results
            #     print('-' * 80)
            #     print(f'{"Ground Truth":25s} | {"Prediction":25s} | Confidence Score & T/F')
            #     log.write(f'{"Ground Truth":25s} | {"Prediction":25s} | {"Confidence Score"}\n')
            #     print('-' * 80)
            #     for gt, pred, confidence in zip(labels[:5], preds[:5], confidence_score[:5]):
            #         if 'Attn' in opt.Prediction:
            #             gt = gt[:gt.find('[s]')]
            #             pred = pred[:pred.find('[s]')]
            #
            #         print(f'{gt:25s} | {pred:25s} | {confidence:0.4f}\t{str(pred == gt)}')
            #         log.write(f'{gt:25s} | {pred:25s} | {confidence:0.4f}\t{str(pred == gt)}\n')
            #     print('-' * 80)

        # save model per 1e+5 iter.

        loss_log = f'[{i}/{opt.num_iter}] Train loss: {loss_avg.val():0.5f}, Elapsed_time: {elapsed_time:0.5f}'
        print(loss_log)
        if (i + 1) % 1e+5 == 0:
            torch.save(
                model.state_dict(), f'./saved_models/{opt.experiment_name}/iter_{i+1}.pth')

**Comment by leigaoxiang** [comment]
and my config as foollow
experiment_name: None-ResNet-BiLSTM-CTC-Seed1111
train_data: /home/lgx/Taxilmdb
valid_data: /home/lgx/Taxilmdb
manualSeed: 1111
workers: 8
batch_size: 32
num_iter: 300000
valInterval: 10
saved_model: 
FT: False
adam: False
lr: 1
beta1: 0.9
rho: 0.95
eps: 1e-08
grad_clip: 5
select_data: ['MJ', 'ST']
batch_ratio: ['0.5', '0.5']
total_data_usage_ratio: 1.0
batch_max_length: 50
imgH: 32
imgW: 280
rgb: False
character:  发票代码号期012345678k9上下车金额￥$-元.:"年月日/时间()#;总A开起午Dd至止G[]xfare人民币J
sensitive: False
PAD: True
data_filtering_off: False
Transformation: None
FeatureExtraction: ResNet
SequenceModeling: BiLSTM
Prediction: CTC
num_fiducial: 20
input_channel: 1
output_channel: 512
hidden_size: 256
num_gpu: 1
num_class: 62

**Comment by ku21fan** [comment]
@leigaoxiang I see..!

Could you check that your terminal can print Chinese characters? 
```
print('发票代码号期012345678k9上下车金额')
```
If you can not print Chinese characters, I guess that you need the language pack which can print Chinese characters.

And Could you uncomment out the whole validation part and then only comment out below part? which shows the predicted result.
In this case, the predicted results would have non-alphabet characters, and that may occur problems.
```
        #     # show some predicted results
        #     print('-' * 80)
        #     print(f'{"Ground Truth":25s} | {"Prediction":25s} | Confidence Score & T/F')
        #     log.write(f'{"Ground Truth":25s} | {"Prediction":25s} | {"Confidence Score"}\n')
        #     print('-' * 80)
        #     for gt, pred, confidence in zip(labels[:5], preds[:5], confidence_score[:5]):
        #         if 'Attn' in opt.Prediction:
        #             gt = gt[:gt.find('[s]')]
        #             pred = pred[:pred.find('[s]')]
        #
        #         print(f'{gt:25s} | {pred:25s} | {confidence:0.4f}\t{str(pred == gt)}')
        #         log.write(f'{gt:25s} | {pred:25s} | {confidence:0.4f}\t{str(pred == gt)}\n')
        #     print('-' * 80)
```

Best

**Comment by leigaoxiang** [comment]
Python 3.6.8 (default, May  7 2019, 14:58:50) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print('发票代码号期012345678k9上下车金额')
发票代码号期012345678k9上下车金额
>>> 

my terminal can print chinese  also i print text variable in CTCLabelConverter encode function ,also can print chinese.

and in test.py validate function i print accuracy
  accuracy = n_correct / float(length_of_data) * 100
    print("hhhhh:",accuracy)
    norm_ED = norm_ED / float(length_of_data)  
the rusult is accuracy can not print

**Comment by leigaoxiang** [comment]
validate function can not program

**Comment by zcswdt** [comment]
> Python 3.6.8 (default, May 7 2019, 14:58:50)
> [GCC 5.4.0 20160609] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> 
> > > > print('发票代码号期012345678k9上下车金额')
> > > > 发票代码号期012345678k9上下车金额
> 
> my terminal can print chinese also i print text variable in CTCLabelConverter encode function ,also can print chinese.
> 
> and in test.py validate function i print accuracy
> accuracy = n_correct / float(length_of_data) * 100
> print("hhhhh:",accuracy)
> norm_ED = norm_ED / float(length_of_data)
> the rusult is accuracy can not print

您好，您在训练过程中，使用了Pad和rgb这两个参数吗？
